var/spack/repos/builtin/packages/py-scikit-learn/package.py

```diff
@@ -35,8 +35,9 @@ class PyScikitLearn(PythonPackage):
     version('0.15.2', 'd9822ad0238e17b382a3c756ea94fe0d')
     version('0.16.1', '363ddda501e3b6b61726aa40b8dbdb7e')
     version('0.17.1', 'a2f8b877e6d99b1ed737144f5a478dfc')
+    version('0.13.1', 'acba398e1d46274b8470f40d0926e6a4')
 
     depends_on('python@2.6:2.8,3.3:')
-    depends_on('py-setuptools',   type='build')
+    depends_on('py-setuptools',   type='build', when='@0.17.1:')
```

Comment:
This is where things get confusing. If you check the `setup.py` file, `setuptools` is only imported for certain `setup.py` commands:
```python
############################################################################### 
# Optional setuptools features                                                  
# We need to import setuptools early, if we want setuptools features,           
# as it monkey-patches the 'setup' function                                     
                                                                                
# For some commands, use setuptools                                             
if len(set(('develop', 'release', 'bdist_egg', 'bdist_rpm',                     
           'bdist_wininst', 'install_egg_info', 'build_sphinx',                 
           'egg_info', 'easy_install', 'upload',                                
            )).intersection(sys.argv)) > 0:                                     
    import setuptools                                                           
    extra_setuptools_args = dict(                                               
            zip_safe=False,  # the package can run out of an .egg file          
            include_package_data=True,                                          
        )                                                                       
else:                                                                           
    extra_setuptools_args = dict()         
```
We run `setup.py build` and `setup.py install`. Since those commands aren't in this list, `setuptools` isn't actually used.

The reason it fails is that we add special setuptools-only flags when building with setuptools.